mitreattack/collections/collection_to_index.py

```diff
@@ -0,0 +1,174 @@
+import argparse
+import os
+from tqdm import tqdm
+import json
+import uuid
+from dateutil.parser import isoparse
+import re
+
+
+class CollectionToIndex:
+    @staticmethod
+    def generate_index(name, description, root_url, files=None, folders=None, bundles=None):
+        """generate a collection index from the input data and return the index as a dict
+        arguments:
+            name (string): the name of the index
+            description (string): the description of the index
+            root_url (string): the root URL where the collections can be found. Specified collection paths will be
+                                appended to this for the collection URL
+            files (string[], optional): collection JSON files to include in the index. Cannot be used with folder
+                                argument
+            folders (string[], optional): folders of collection JSON files to include in the index. Cannot be used with
+                                files argument. Will only match collections that end with a version number
+            bundles (stix bundle[], optional): array of stix bundle objects to include in the index
+            output (string, optional): filename for the generated collection index file
+        """
+        if files and folders:
+            print("cannot use both files and folder at the same time, please use only one argument at a time")
```

Comment:
This should also exclude the bundles input, right? I believe this code is intended to make sure the various input argument types are mutually exclusive in use.

Reply:
The method doc comment would need to clarify this as well, currently only files and folders mention the mutually exclusiveness of their use.

Reply:
Technically, I think the code as is supports either files/folders and bundles, but that's rather confusing. I'll streamline.